ionic_user_interface/src/views/UploadRoute.vue

```diff
@@ -0,0 +1,313 @@
+      selectedCountryIso3.value = '';
+      selectedGymLocation.value = '';
+    };
+
+    const onCountrySelect = async (country: Country) => {
+      if (country) {
+        selectedCountryIso3.value = country.iso3;
+        gymLocationList.value = await getGyms(country.iso3);
+      } else {
+        reset();
+      }
+    };
+
+    const isValidRouteName = (routeName: string): boolean => {
+      return routeName.length > 0 && routeName.length <= 30 && asciiPattern.test(routeName);
+    };
+
+    const showErrorMsg = (errorMsg: string): void => {
+      msgBoxColor.value = 'danger';
+      msgBox.value?.showMsg(errorMsg);
+      content.value?.$el.scrollToTop(400);
+    };
+
+    const onSubmit = throttle(async (event: Event): Promise<boolean> => {
+      event.preventDefault();
+      msgBox.value?.close();
+
+      if (!isValidRouteName(routeNameText.value)) {
+        showErrorMsg('Route name has to be 0-30 ASCII characters');
+        return false;
+      }
+
+      if (!userHasSelectedCountry.value || selectedGymLocation.value === '') {
+        showErrorMsg('Please select a country and gym');
+        return false;
+      }
+
+      if (expiryTime.value === '') {
+        showErrorMsg('Please select an expiry date');
+        return false;
+      }
+      const routeImageBlob = await fetch(routeImage.value).then((res) => res.blob());
+      const formData = new FormData();
+      formData.append('countryCode', selectedCountryIso3.value);
+      formData.append('expiredTime', new Date(expiryTime.value).toISOString());
+      formData.append('gymLocation', selectedGymLocation.value);
+      formData.append('ownerGrade', gradeNumber.value.toString());
+      formData.append('routePhoto', routeImageBlob);
+      formData.append('routeName', routeNameText.value);
+
+      axios
+        .post(process.env.VUE_APP_ROUTE_ENDPOINT_URL + '/route/new', formData, {
+          headers: {
+            'Content-Type': 'multipart/form-data',
+            Authorization: `Bearer ${getAccessToken().value}`,
+          },
+        })
+        .then((response) => {
+          console.log(response);
+          if (response.data.Message === 'Create route success') {
+            toastController
+              .create({
+                header: 'Route has been uploaded successfully',
+                position: 'bottom',
+                color: 'success',
+                duration: 3000,
+                buttons: [
+                  {
+                    text: 'Close',
+                    role: 'cancel',
+                  },
+                ],
+              })
+              .then((toast) => {
+                toast.present();
+              });
+
+            router.push('/home');
+          } else {
+            showErrorMsg('Unable to create route: ' + response.data.Message);
+          }
+        })
+        .catch((error) => {
+          if (error.response) {
+            if (error.response.data.Message === 'CodeMismatchException') {
```

Comment:
This is a mistake right?

Reply:
lmao forgot to change the error handling parts